client/signup/steps/site-options/index.tsx

```diff
@@ -20,10 +21,24 @@ interface Props {
 export default function SiteOptionsStep( props: Props ): React.ReactNode {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
-	const headerText = translate( "First, let's give your blog a name" );
 	const { stepName, signupDependencies, goToNextStep } = props;
 	const { siteTitle, tagline } = signupDependencies;
 
+	const headerText =
+		'store-options' === stepName
+			? translate( "First, let's give your store a name" )
+			: translate( "First, let's give your blog a name" );
+
+	const headerImage = 'store-options' === stepName ? intentImageUrl : siteOptionsImage;
+
+	const siteTitleLabel =
+		'store-options' === stepName ? translate( 'Store name' ) : translate( 'Blog name' );
+
+	const taglineExplanation =
+		'store-options' === stepName
+			? translate( 'In a few words, explain what your store is about.' )
+			: translate( 'In a few words, explain what your blog is about.' );
+
```

Comment:
This is somewhat picky but it feels like this section might get unwieldy if we add further custom intents in the future. Possibly we could refactor this a bit to be more  extensible.

(Off the top of my head, my initial thought is to lean towards something like 

```
const getIntentText = () => {
    switch (stepName) {
        case 'store-options':
            return {
                headerText: translate( "First, let's give your store a name" ),
                headerImage: intentImageUrl,
                siteTitleLabel: translate( 'Store name' ),
                taglineExplanation: translate( 'In a few words, explain what your store is about.' )
            };
        default:
            return {
                headerText: translate( "First, let's give your blog a name" ),
                headerImage: siteOptionsImage,
                siteTitleLabel: translate( 'Blog name' ),
                taglineExplanation: translate( 'In a few words, explain what your blog is about.' )
            }
        }
    }
}
```
But this is just a rough thought. Just generally, though, the long string of ternaries feels like something that would be difficult to further modify later on.

Reply:
Agreed!